### PR TITLE
Upgrade binutils to 2.41 to fix CVEs

### PR DESCRIPTION
**Pull request by miz060**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Upgrade binutils from 2.38 to 2.41 to fix CVE-2022-35206, CVE-2022-35205, CVE-2022-48063, CVE-2023-1972, CVE-2022-47007, CVE-2022-47008, CVE-2022-47010, CVE-2022-47011

###### Change Log  

- Upgrade binutils from 2.38 to 2.41

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-35206
- https://nvd.nist.gov/vuln/detail/CVE-2022-35205
- https://nvd.nist.gov/vuln/detail/CVE-2022-48063
- https://nvd.nist.gov/vuln/detail/CVE-2023-1972
- https://nvd.nist.gov/vuln/detail/CVE-2022-47007
- https://nvd.nist.gov/vuln/detail/CVE-2022-47008
- https://nvd.nist.gov/vuln/detail/CVE-2022-47010
- https://nvd.nist.gov/vuln/detail/CVE-2022-47011

###### Test Methodology

- BuddyBuild: [654186](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=654186&view=results)

### CODE REVIEW

**Review by xordux — APPROVED** [review]

**Review by jslobodzian — CHANGES_REQUESTED** [review]
Not sure we can take a bump to binutils like this.  It frequently causes other issue.

Please have Andrew take a look

File: SPECS/binutils/linker-script-readonly-keyword-support.patch [hunk 1]
```diff
@@ -1,156 +0,0 @@
-From 6b86da53d5ee2022b9065f445d23356190380746 Mon Sep 17 00:00:00 2001
```
Comment (anphel31) [inline SPECS/binutils/linker-script-readonly-keyword-support.patch:1]: This file still needs to be here. The bootstrap build of binutils is still version 2.37 and it requires it

**Comment by anphel31** [comment]
We will need to run a full build to validate this change.
Would also be best to upgrade the binutils 2.37 in the bootstrap toolchain.

**Comment by trungams** [comment]
closing in favor of https://github.com/microsoft/azurelinux/pull/11082

**Pull request closed without merging**